src/table.rs

```diff
@@ -61,40 +62,351 @@ impl Table {
+        }
+    }
+
+    pub fn get_river_string_secret(&self) -> String {
+        match self.river {
+            None => { "None".to_string() }
+            Some(card) => {
+                match &self.table_state {
+                    River => { card.to_string() }
+                    _ => { "Hidden".to_string() }
+                }
+            }
+        }
+    }
+
+    /// Returns the number of players
     pub fn get_player_count(&self) -> usize {
         self.players.len()
     }
 
+    /// Takes an action, could be recursive if the table needs no input
+    pub fn take_action(&mut self, hand_action: HandAction) {
+        // If the game is over print out a message, and do not take any actions
+        if self.is_game_over() {
+            println!("Game is over! Results are included below:\n{}", self.get_results());
+            return;
+        }
+        // Make sure the player is active, or increment and go to the next player
+        if let PlayerState::Active(active) = self.get_current_player().player_state {
+            self.take_provided_action(hand_action, active);
+        } else {
+            panic!("Tried to take an action on a dead player");
+        }
+        // If there is only 1 alive player evaluate the winner
+        if self.get_alive_player_count() == 1 {
+            self.pick_winner();
+            return;
+        }
+        // The round of betting is over
+        if self.is_betting_over() {
+            // The showdown is occurring, pick the winner
+            if self.table_state == River {
+                self.pick_winner();
+                return;
+            }
+            // Move to the next betting stage
+            self.table_state.next_stage();
+            // Reset the turn to the next person alive past the deal index
+            self.current_player_index = self.dealer_button_index;
+            self.update_current_player_index_to_next_active();
+            // set everyone to not have a turn yet
+            for player in &mut self.players {
+                player.has_had_turn_this_round = false;
+            }
+        }
+    }
+
+    fn take_provided_action(&mut self, hand_action: HandAction, active_state: ActiveState) {
+        let difference = self.get_max_bet() - active_state.current_bet;
+        // Now check how to advance the hand
+        match hand_action {
+            HandAction::Fold => {
+                self.get_current_player().fold();
+            }
+            HandAction::Check => {
+                if difference == 0 {
+                    self.get_current_player().bet(0);
+                } else {
+                    self.get_current_player().fold();
+                }
+            }
+            HandAction::Call => {
+                self.pot += self.get_current_player().bet(difference);
+            }
+            HandAction::Raise(raise_amount) => {
+                self.pot += self.get_current_player().bet(difference + raise_amount);
```

Comment:
Doesn't need to be in this MR, but might want to add a TODO here if this is where the max bet logic will go

Reply:
Added